Remove WKPDFView

aprotyas opened the pull request:
#### 4e0dd1ff1a4ca32b1e1743411884613c09470589
<pre>
Remove WKPDFView
<a href="https://bugs.webkit.org/show_bug.cgi?id=296654">https://bugs.webkit.org/show_bug.cgi?id=296654</a>
<a href="https://rdar.apple.com/157053061">rdar://157053061</a>

Reviewed by Megan Gardner, Aditya Keerthi, and Tim Horton.

WKPDFView is no longer supported on trunk given that we&apos;ve moved on with
UnifiedPDFPlugin on iOS family too. In this patch, we remove WKPDFView.
This also lets us perform adjacent cleanup, as detailed below:

1. Remove the USE_WKPDFVIEW and HAVE_SETUSEIOSURFACEFORTILES macros. The
   latter was used to gate -setUseIOSurfaceForTiles: in WKPDFView.
2. Remove PDFHVC declarations in PDFKitSPI.h.
3. Remove PDFHVC soft link support for tvOS.
4. Remove many methods from the WKWebViewContentProvider interface. This
   includes optional methods that were only implemented by WKPDFView, as
   well as web_requiresCustomSnapshotting, which was only a non-constant
   for WKPDFView. Removal of the custom snapshotting getter allowed us
   to some snapshot code.
5. Greatly simplify -_isDisplayingPDF to no longer consult either the
   presence of custom content providers or the UnifiedPDF feature status.
6. Remove -_tryToHandleKeyEventInCustomContentView and all paths that
   called it since WKUSDPreviewView did not use web_handleKeyEvent. This
   also allowed us to remove the presses[Began|Changed|Ended|Cancelled]:
   implementations on WKWebView.
7. Similarly, simplify the UIScrollViewDelegate implementations on
   WKWebView because WKUSDPreviewView did not do any custom scrolling.
8. Simplify [_data|suggestedFilename]ForDisplayedPDF to unconditionally
   return nil. These methods have been deprecated and return nil when
   Unified PDF is enabled, anyway.
9. No longer register WKPDFView as a PDF content provider in the content
   provider registry.

* LayoutTests/fast/events/ios/pdf-modifer-key-down-crash.html:
* LayoutTests/fast/replaced/encrypted-pdf-as-object-and-embed.html:
* LayoutTests/fast/replaced/pdf-iframe-load-crash.html:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:
* Source/WebKit/Shared/Cocoa/PDFKitSoftLink.h:
* Source/WebKit/Shared/Cocoa/PDFKitSoftLink.mm:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView takeSnapshotWithConfiguration:completionHandler:]):
(-[WKWebView _isDisplayingPDF]):
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView scrollViewWillBeginZooming:withView:]):
(-[WKWebView scrollViewDidScroll:]):
(-[WKWebView scrollViewDidZoom:]):
(-[WKWebView scrollViewDidEndZooming:withView:atScale:]):
(-[WKWebView _dataForDisplayedPDF]):
(-[WKWebView _suggestedFilenameForDisplayedPDF]):
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
(-[WKWebView _snapshotRectAfterScreenUpdates:rectInViewCoordinates:intoImageOfWidth:completionHandler:]):
(-[WKWebView _tryToHandleKeyEventInCustomContentView:]): Deleted.
(-[WKWebView pressesBegan:withEvent:]):
(-[WKWebView pressesChanged:withEvent:]):
(-[WKWebView pressesEnded:withEvent:]):
(-[WKWebView pressesCancelled:withEvent:]):
* Source/WebKit/UIProcess/Cocoa/WKWebViewContentProvider.h:
* Source/WebKit/UIProcess/Cocoa/WKWebViewContentProviderRegistry.mm:
(-[WKWebViewContentProviderRegistry initWithConfiguration:]):
* Source/WebKit/UIProcess/ios/WKPDFView.h: Removed.
* Source/WebKit/UIProcess/ios/WKPDFView.mm: Removed.
* Source/WebKit/UIProcess/ios/WKUSDPreviewView.mm:
(+[WKUSDPreviewView web_requiresCustomSnapshotting]): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/QuickLook.mm:
(runTest):
(TEST(QuickLook, AllowResponseAfterLoadingPreview)):
(TEST(QuickLook, CancelResponseAfterLoadingPreview)):
(TEST(QuickLook, DownloadResponseAfterLoadingPreview)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKNavigationResponse.mm:
(TEST(WebKit, WKNavigationResponsePDFType)):

Canonical link: <a href="https://commits.webkit.org/298088@main">https://commits.webkit.org/298088@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dcb092918a256fbe1eb9cb98cfba5f44c2dde684

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114166 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33916 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24375 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120333 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64915 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5fee71da-678b-4783-8fd2-379536c9c69f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116055 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34543 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42477 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86767 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/989d5197-9b7f-47fc-840a-3413ce6943f6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117114 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27508 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102542 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67155 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26692 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20668 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64028 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/106577 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96880 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20785 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123551 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/112729 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41188 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30707 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95601 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41566 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98745 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95384 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24313 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40521 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18341 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/37301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41068 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46575 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136928 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40681 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36657 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43983 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42433 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f36db06f9a50e61d744d19314f5a6b031f1ad7d4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48739

pxlcoder commented on Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
```diff
@@ -4456,16 +4398,12 @@ - (BOOL)_allowsViewportShrinkToFit
 - (NSData *)_dataForDisplayedPDF
 {
-    if (![self _isDisplayingPDF])
-        return nil;
-    return [_customContentView web_dataRepresentation];
+    return nil;
 }
 
 - (NSString *)_suggestedFilenameForDisplayedPDF
```
Should we file some bugs to remove this SPI?

aprotyas replied:
Yes, good idea.

aprotyas replied:
Remove -_dataForDisplayedPDF and -_suggestedFilenameForDisplayedPDF SPIs
https://bugs.webkit.org/show_bug.cgi?id=296719

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dcb092918a256fbe1eb9cb98cfba5f44c2dde684. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48739

aprotyas commented:
Thank you for the reviews!

aprotyas committed:
Remove WKPDFView
https://bugs.webkit.org/show_bug.cgi?id=296654
rdar://157053061

Reviewed by Megan Gardner, Aditya Keerthi, and Tim Horton.

WKPDFView is no longer supported on trunk given that we've moved on with
UnifiedPDFPlugin on iOS family too. In this patch, we remove WKPDFView.
This also lets us perform adjacent cleanup, as detailed below:

1. Remove the USE_WKPDFVIEW and HAVE_SETUSEIOSURFACEFORTILES macros. The
   latter was used to gate -setUseIOSurfaceForTiles: in WKPDFView.
2. Remove PDFHVC declarations in PDFKitSPI.h.
3. Remove PDFHVC soft link support for tvOS.
4. Remove many methods from the WKWebViewContentProvider interface. This
   includes optional methods that were only implemented by WKPDFView, as
   well as web_requiresCustomSnapshotting, which was only a non-constant
   for WKPDFView. Removal of the custom snapshotting getter allowed us
   to some snapshot code.
5. Greatly simplify -_isDisplayingPDF to no longer consult either the
   presence of custom content providers or the UnifiedPDF feature status.
6. Remove -_tryToHandleKeyEventInCustomContentView and all paths that
   called it since WKUSDPreviewView did not use web_handleKeyEvent. This
   also allowed us to remove the presses[Began|Changed|Ended|Cancelled]:
   implementations on WKWebView.
7. Similarly, simplify the UIScrollViewDelegate implementations on
   WKWebView because WKUSDPreviewView did not do any custom scrolling.
8. Simplify [_data|suggestedFilename]ForDisplayedPDF to unconditionally
   return nil. These methods have been deprecated and return nil when
   Unified PDF is enabled, anyway.
9. No longer register WKPDFView as a PDF content provider in the content
   provider registry.

* LayoutTests/fast/events/ios/pdf-modifer-key-down-crash.html:
* LayoutTests/fast/replaced/encrypted-pdf-as-object-and-embed.html:
* LayoutTests/fast/replaced/pdf-iframe-load-crash.html:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:
* Source/WebKit/Shared/Cocoa/PDFKitSoftLink.h:
* Source/WebKit/Shared/Cocoa/PDFKitSoftLink.mm:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView takeSnapshotWithConfiguration:completionHandler:]):
(-[WKWebView _isDisplayingPDF]):
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView scrollViewWillBeginZooming:withView:]):
(-[WKWebView scrollViewDidScroll:]):
(-[WKWebView scrollViewDidZoom:]):
(-[WKWebView scrollViewDidEndZooming:withView:atScale:]):
(-[WKWebView _dataForDisplayedPDF]):
(-[WKWebView _suggestedFilenameForDisplayedPDF]):
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
(-[WKWebView _snapshotRectAfterScreenUpdates:rectInViewCoordinates:intoImageOfWidth:completionHandler:]):
(-[WKWebView _tryToHandleKeyEventInCustomContentView:]): Deleted.
(-[WKWebView pressesBegan:withEvent:]):
(-[WKWebView pressesChanged:withEvent:]):
(-[WKWebView pressesEnded:withEvent:]):
(-[WKWebView pressesCancelled:withEvent:]):
* Source/WebKit/UIProcess/Cocoa/WKWebViewContentProvider.h:
* Source/WebKit/UIProcess/Cocoa/WKWebViewContentProviderRegistry.mm:
(-[WKWebViewContentProviderRegistry initWithConfiguration:]):
* Source/WebKit/UIProcess/ios/WKPDFView.h: Removed.
* Source/WebKit/UIProcess/ios/WKPDFView.mm: Removed.
* Source/WebKit/UIProcess/ios/WKUSDPreviewView.mm:
(+[WKUSDPreviewView web_requiresCustomSnapshotting]): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/QuickLook.mm:
(runTest):
(TEST(QuickLook, AllowResponseAfterLoadingPreview)):
(TEST(QuickLook, CancelResponseAfterLoadingPreview)):
(TEST(QuickLook, DownloadResponseAfterLoadingPreview)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKNavigationResponse.mm:
(TEST(WebKit, WKNavigationResponsePDFType)):

Canonical link: https://commits.webkit.org/298088@main
